src/lib/containers/table/SynapseTable.tsx

```diff
@@ -184,9 +185,13 @@ export default class SynapseTable extends React.Component<
   }
   componentDidUpdate(prevProps: QueryWrapperChildProps & SynapseTableProps, prevState: Readonly<SynapseTableState>) {
     // PORTALS-2081: if the data changed, then get the new entity headers
-    if (!eq(prevProps.data, this.props.data) || !eq(prevState.isEntityView, this.state.isEntityView)) {
+    if (!eq(prevProps.data, this.props.data)) {
       this.getEntityHeadersInData(false)
+    } else if (!eq(prevState.isEntityView, this.state.isEntityView)) {
+      // if we determined that this is an entity view, force refresh the entity headers (for Views we need to get the rowIds!)
+      this.getEntityHeadersInData(true)
```

Comment:
If we determined that this is an EntityView, then we need to get EntityHeaders for the row IDs as well, so force it!

Reply:
Should it be this?
```ts
     } else if (this.state.isEntityView && !eq(prevState.isEntityView, this.state.isEntityView)) {
```

Seems like if we go from View -> TableEntity we shouldn't do this, but maybe I don't fully understand this.